src/components/bulk-email-tool/bulk-email-form/BulkEmailForm.jsx

```diff
@@ -47,19 +64,26 @@ export default function BulkEmailForm(props) {
   };
 
   const onSubjectChange = (event) => setSubject(event.target.value);
+  const onDateTimeChange = (event) => {
+    const next = { [event.target.name]: event.target.value };
+    setDateTime((prev) => ({ ...prev, ...next }));
+  };
```

Comment:
I don't think I'm following what this is for? Sorry to be daft, do you mind explaining what the purpose of `onDateTimeChange` is for?

Reply:
I'm using a single piece of state and a single `onChange` function to capture the date and time (they come from separate inputs). This could have been inline on the component, but I extracted it out because I'm using it in two places. `event.target` will either be the date _or_ the time input when this executes.

the reason why I'm defining `next` here is because setting state is async, and the event target may no longer be the present when `setDateTime()` is executing. 

in practice:
- The user either inputs the date or the time in one of the input fields
- `onDateTimeChange` executes, capture the value in the `next` object
- we spread the new value in `next` over the previous object in the `dateTime` state. This preserves any value not being overwritten (like if the date is already present and the user edits the time).